cloudinit/cmd/devel/logs.py

```diff
@@ -7,49 +7,25 @@
 """Define 'collect-logs' utility and handler to include in cloud-init cmd."""
 
 import argparse
+import itertools
 import logging
 import os
 import pathlib
-import shutil
+import stat
 import subprocess
 import sys
 from datetime import datetime, timezone
-from typing import List, NamedTuple, Optional, cast
+from typing import Any, Dict, Iterator, List, NamedTuple, Optional, cast
 
 from cloudinit import log
-from cloudinit.cmd.devel import read_cfg_paths
 from cloudinit.stages import Init
 from cloudinit.subp import ProcessExecutionError, subp
 from cloudinit.temp_utils import tempdir
-from cloudinit.util import (
-    chdir,
```

Comment:
I know that you didn't write this, but we shouldn't need this helper at all. This is the only callsite of `util.chdir()`, yet `tar` (including busybox tar) has `-C` which can switch directories before running the operation.

Reply:
Alas, it is still used in bbdeb.

Reply:
> Alas, it is still used in bbdeb.

I'm pretty strongly opposed to shipping dead code in cloud-init that exists only for use in external utilities, but I'm also fine with saying that it's not worth dealing with right now.